docs/proposals/image-spec.md

```diff
@@ -0,0 +1,480 @@
+        image: coreos.com/etcd:v2.2.0
+      - name: flannel
+        image: coreos.com/flannel
+    status:
+      containerStatuses:
+      - containerID: rkt://foo
+        image: coreos.com/etcd
+        ...
+      - containerID: rkt://bar
+        image: coreos.com/flannel
+        ...
+    ```
+
+    *Note that an old client cannot know from this specification what the image format type is.
+    This also means that for old servers, the image spec will be translated to a Docker-like image string,
+    and if the Docker image does not exist, the pod cannot be created based on this spec. However, users
+    SHOULD NOT send such requests to servers using the old API version, as old servers will not recognize
+    image formats other than Docker.*
+
+    looks to new clients like:
+    
+    ```yaml
+    apiVersion: v1
+    kind: Pod
+    metadata:
+    name: test-pod
+    spec:
+      containers:
+      - name: etcd
+        image: coreos.com/etcd:v2.2.0
+        imageSpec:
+          aci:
+            name: coreos.com/etcd
+            tag: v2.2.0
+      - name: flannel
+        image: coreos.com/flannel
+        imageSpec:
+          aci:
+            name: coreos.com/flannel
+    status:
+      containerStatuses:
+      - containerID: rkt://foo
+        image: coreos.com/etcd
+        ...
+      - containerID: rkt://bar
+        image: coreos.com/flannel
+        ...
+    ```
+
+## Kubelet/container runtime changes
+
+1. The [generic image puller](https://github.com/kubernetes/kubernetes/blob/v1.2.0-alpha.4/pkg/kubelet/container/image_puller.go#L82)
+   should be changed to use `ImageSpec`.
+
+2. Image related functions in the [container runtime interface](https://github.com/kubernetes/kubernetes/blob/v1.2.0-alpha.4/pkg/kubelet/container/runtime.go#L75),
+   (including `PullImage`, `IsImagePresent`, `RemoveImage`) should be changed to use `ImageSpec`.
+   Besides, these functions should return an error if the container runtime doesn't support the given image format.
+
+## Testing
+
+1. Add test cases to verify the API semantics proposed here, e.g. One and only one image format should be present in
+   the `ImageSpec`, etc.
+
+2. Add backward compatibility tests to verify the compatibility by converting the objects between internal API and
+   versioned APIs. Examples above or similar examples should be used as the test cases.
+
+3. Add unit tests in kubelet package to verify the changes made in kubelet/container runtime.
+
+4. E2E test cases should be added to verify things such as:
+   - `ImageSpec` and `Image` are mutually exclusive, and at least one of them should be specified.
+   - Container runtime should reject unsupported image format.
+   - ACI should be supported when using rkt as a runtime.
+
+
+## Future work
+
+Eventually, the scheduler should be aware of the image format, and it should prevent a pod being scheduled
+to a node whose container runtime doesn't support the specified image format. This requires:
```

Comment:
This feature is only needed if we support multiple runtimes within the same cluster right?